src/main/scala/Bag.scala

```diff
@@ -0,0 +1,39 @@
+import scala.util.Random
+
+object Bag {
+
+  var letterDistribution: Map[Char, Int] = Map(
+    'A' -> 9,
+    'B' -> 2,
+    'C' -> 2,
+    'D' -> 4,
+    'E' -> 12,
+    'F' -> 2,
+    'G' -> 3,
+    'H' -> 5,
+    'I' -> 9,
+    'J' -> 1,
+    'K' -> 1,
+    'L' -> 4,
+    'M' -> 2,
+    'N' -> 6,
+    'O' -> 8,
+    'P' -> 2,
+    'Q' -> 1,
+    'R' -> 6,
+    'S' -> 4,
+    'T' -> 6,
+    'U' -> 4,
+    'V' -> 2,
+    'W' -> 2,
+    'Y' -> 2,
+    'X' -> 1,
+    'Z' -> 1,
+  )
+
+  def initialise(): List[Tile] = {
+    def dupeLetters(letter: (Char, Int)): List[Tile] = List.fill(letter._2)(Tile(letter._1))
+    Random.shuffle(letterDistribution.toList.flatMap(dupeLetters))
```

Comment:
by having a non referentially transparent function (Random.shuffle does something different each time you call it) it makes this function non deterministic and thus not fully unit testable.

A common technique is to push any non determinism as high up in the code as possible, so ideally the top level function would just be calling the deterministic main part of the program, and then running all the side effects/impure parts.

In this case that would be to always return the list in order, and shuffle it in the parent function.

Reply:
Makes sense! Could complete the test and also learnt that Maps do not preserver order!